--- a/drivers/tty/serial/ar933x_uart.c
+++ b/drivers/tty/serial/ar933x_uart.c
@@ -199,6 +199,25 @@ static void ar933x_uart_get_scale_step(u
 	unsigned int tscale;
 	long min_diff;
 
+#if defined(CONFIG_ATH79_MACH_LININO_YUN) || defined(CONFIG_ATH79_MACH_LININO_ONE)
+	/* Fix timing issues on Atmega32U4 w/16Mhz oscillator */
+	if (baud == 115200) {
+		*scale = 0x000C;
+		*step  = 0x2000;
+		return;
+	}
+	if (baud == 250000 || baud == 230400) {
+		*scale = 0x0017;
+		*step  = 0x7AE0;
+		return;
+	}
+	if (baud == 500000) {
+		*scale = 0x000B;
+		*step  = 0x7AE0;
+		return;
+	}
+#endif
+
 	*scale = 0;
 	*step = 0;
 
